packages/utils/odsp-doclib-utils/src/odspAuth.ts

```diff
@@ -104,25 +105,59 @@ export async function fetchTokens(
 		client_secret: clientConfig.clientSecret,
 		...credentials,
 	};
-	const result = await unauthPostAsync(
+	const response = await unauthPostAsync(
 		getFetchTokenUrl(server),
 		new URLSearchParams(body), // This formats the body like a query string which is the expected format
 	);
-	const tokens = await result.json();
-	const accessToken = tokens.access_token;
-	const refreshToken = tokens.refresh_token;
+
+	const parsedResponse = await response.json();
+	const accessToken = parsedResponse.access_token;
+	const refreshToken = parsedResponse.refresh_token;
 
 	if (accessToken === undefined || refreshToken === undefined) {
-		throwOdspNetworkError(
-			// pre-0.58 error message: unableToGetAccessToken
-			"Unable to get access token",
-			tokens.error === "invalid_grant" ? 401 : result.status,
-			result,
-		);
+		try {
+			throwOdspNetworkError(
+				// pre-0.58 error message: unableToGetAccessToken
+				"Unable to get access token.",
+				parsedResponse.error === "invalid_grant" ? 401 : response.status,
+				response,
+			);
+		} catch (error) {
+			if (isFluidError(error) && isAccessTokenError(parsedResponse)) {
+				error.addTelemetryProperties({
+					error: parsedResponse.error,
```

Comment:
you will be overwriting error property that way (which is the value of thrown error.message).

Reply:
good catch. changed prop name.